dbt_sugar/core/task/doc.py

```diff
@@ -56,10 +58,123 @@ def run(self) -> int:
+            and the boolean indicating if the schema.yml exists.
+        """
+        for root, _, files in os.walk(self.repository_path):
+            for file in files:
+                # TODO: Check how to avoid using target to discriminate compiled files in DBT.
+                if file == f"{model_name}.sql" and "target" not in root:
+                    path_file = Path(os.path.join(root, "schema.yml"))
+                    if path_file.is_file():
+                        return path_file, True
+                    else:
+                        return path_file, False
+        return None, False
+
+    def orchestrator_document_model(self, model_name: str, columns_sql: List[str]) -> None:
+        """
+        Orchestrator to fully document a model will:
+
+            - Create or update the columns from the database into the shema.yaml.
+            - Gives the user the posibility to change the model description.
+            - Gives the user the posibility to document the undocumented columns.
+            - Updates all the new columns definitions in all DBT.
+
+        Args:
+            model_name (str): model name that we want to document.
+            columns_sql (List[str]): list of the columns of the model in the database.
+        """
+        content = None
+        path, exists = self.find_model_in_dbt(model_name)
+        if not path:
+            logger.info("Not able to find the model in the repository.")
+            return
+        if exists:
+            content = open_yaml(path)
+        content = self.process_model(content, model_name, columns_sql)
+        content = self.change_model_description(content, model_name)
+        save_yaml(path, content)
+
+        not_documented_columns = self.get_not_documented_columns(content, model_name)
+        self.document_columns(not_documented_columns)
+
+        self.update_column_descriptions(self.columns_to_update)
+
+    def document_columns(self, columns: Dict[str, str]) -> None:
+        """
+        Method to document the columns from a model.
+
+        Will ask the user which columns want to document and the new definition
+
+        Args:
+            columns (Dict[str, str]): Dict of columns with the column name
+            and the actual column description to document.
+        """
+        columns_names = list(columns.keys())
+        for i in range(0, len(columns_names), 2):
+            final_index = i + 2
+            undocumented_columns_payload: List[Mapping[str, Any]] = [
+                {
+                    "type": "checkbox",
+                    "name": "cols_to_document",
+                    "choices": columns_names[i:final_index],
+                    "message": "Select the columns you want to document.",
+                }
+            ]
+            user_input = UserInputCollector(
+                "undocumented_columns", undocumented_columns_payload
+            ).collect()
+            cast(Mapping[str, str], user_input)
```

Comment:
Depending what your answer is to this message: https://discord.com/channels/752101657218908281/789211881117057064/800341854809489408

You will probably want to do the following:
```python
user_input_dict = cast(Dict[str, str], user_input)
self.columns_to_update.update(user_input_dict)
```